docs/metrics/wer-cer-mer.md

```diff
@@ -0,0 +1,196 @@
+    <b>Character-level Substitutions</b>: <br>
+    <code>Amidst the emerald <b><big>sh</big></b>adow, butterflies <b><big>whistl</big></b>ed secrets <b><big>o</big></b>n the breeze.</code> <br>
+
+    In the above example, there are 3 word-level substitutions and 9 character-level substitutions.
+
+??? example "Deletion Errors"
+    Deletions occur when a candidate text is missing a word or sequence of words from the reference text.
+
+    <b> Example: </b> <br>
+    Reference: `Amidst the emerald meadow, butterflies whispered secrets in the breeze.` <br>
+    Candidate: `Amidst the emerald meadow, butterflies whispered.` <br>
+
+    <b>Word-level Deletions</b>: <br>
+    <code>Amidst the emerald meadow, butterflies whispered <u><big>secrets in the breeze</big></u>.</code> <br>
+    <b>Character-level Deletions</b>: <br>
+    <code>Amidst the emerald meadow, butterflies whispered <u><big>secrets in the breeze</big></u>.</code> <br>
+
+    In the above example, there are 4 word-level deletions and 18 character-level deletions.
+
+??? example "Insertion Errors"
+    Insertions occur when a candidate text contains an extra word or sequence of words that is not present in the reference text.
+
+    <b> Example: </b> <br>
+    Reference: `Amidst the emerald meadow, butterflies whispered secrets in the breeze.` <br>
+    Candidate: `Amidst the emerald meadow, butterflies whispered ethereal secrets in the breeze.` <br>
+
+    <b>Word-level Insertions</b>: <br>
+    <code>Amidst the emerald meadow, butterflies whispered <i><big>ethereal</big></i> secrets in the breeze.</code> <br>
+    <b>Character-level Insertions</b>: <br>
+    <code>Amidst the emerald meadow, butterflies whispered <i><big>ethereal</big></i> secrets in the breeze.</code> <br>
+
+    In the above example, there is 1 word-level insertion and 8 character-level insertions.
+
+Formally, it is defined as the rate of word-level errors in a candidate text.
+
+$$
+\text{WER} = \frac{\text{Substitutions} + \text{Deletions} + \text{Insertions}}{\text{# of Words in Reference}}
+$$
+
+### Example
+Let's calculate the word error rate between the following reference and candidate texts:
+
+| <b>Reference</b> | <b>Candidate</b> |
+| | |
+|  `The bard sang the ancient melodies of nature, transforming tranquil meadows into sonnets for enhanced soulful grace.` | `The poetic bard echoed the ancient melodies, transcending meadows into sonnets for soulful grace.` |
+
+??? example "Step 1. Count Errors"
+    Highlighting the <b><big>substitution</big></b>, <u><big>deletion</big></u>, and <i><big>insertion</big></i> errors, we can count each type of error:
+
+    <code>
+    The <i><big>poetic</big></i> bard <b><big>echoed</big></b> the ancient melodies <u><big>of nature</big></u>, <b><big>transcending</big></b> <u><big>tranquil</big></u> meadows into sonnets for <u><big>enhanced</big></u> soulful grace.
```

Comment:
This text isn't reference nor candidate.  I understand that we needed this to show both insertion and deletion.  Instead of just having one text which is a mix of both insertion and deletion, should we highlight substitution, deletion and insertion on two separate texts: reference and candidate?

Reply:
Hmm, I've thought about this (kinda like a diffchecker) and I think it would be pretty useful to show both. However, it would be harder for readers to count/visualize the # of errors if there are substitution errors that appear in both candidate and reference. Should we keep it as it is or should we show it in a diffchecker-style view?

![image](https://github.com/kolenaIO/kolena/assets/31229051/1d61e9fe-1906-4736-813c-93e82754fed4)